README.rst

```diff
@@ -409,6 +413,10 @@ The following flags are accepted:
 |                                                                                                                                                         |
 | Gazelle will not process packages outside this directory.                                                                                               |
 +----------------------------------------------------------------------------------------------------------+----------------------------------------------+
+| :flag:`-to_macro macroFile%defName`                                                                      |                                              |
++----------------------------------------------------------------------------------------------------------+----------------------------------------------+
+| Tells Gazelle to write repository rules into a .bzl macro function rather than the WORKSPACE file.                                                      |
```

Comment:
It should probably be pointed out here that Gazelle won't see repositories declared in the macro when updating build files. That means that it won't use custom repository names, and it will go out to the network when trying to determine repository roots.

This is a solvable problem of course, but let's leave it for another time.

Reply:
Hmmm, this was an oversight on my part. We are pretty dependent on the functionality you mentioned, so I would like to get this done sooner than later.

From a quick glance, it looks like one way to get this done would be to add a new flag named something like `-go_repositories_macro=path`.  If the flag is set we call `LoadMacroFile` from `newFixUpdateConfiguration`. We then call `FixWorkspace` and `FixLoads` for both the `macro` and `workspace` files, and also pass both files to `repo.ListRepositories`.

WDYT?

Reply:
#132 describes a `# gazelle:repository` directive that would declare this information in WORKSPACE without requiring command line arguments. That would be a useful escape hatch. I just opened #483, describing a more complete solution. In WORKSPACE, we could have a directive that tells Gazelle to read a macro in a separate .bzl file. Bonus points if we can update existing rules in that macro without needing the `-to_macro` flag.

Until those are implemented, we should mention this problem here, since the problem won't be obvious to people trying this out.

Reply:
I think we can simplify this problem a bit to start with. We can have a default bzl file like `//go_repositories.bzl` unless it's overridden by some the Gazelle directive proposed in #483. Then `--to_macro` becomes a boolean flag, and write `go_repository` rules to `//go_repositories.bzl` when the flag is set. `update` and `fix` will first check if there is a `//go_repositories.bzl`. If so, read those rules first. Then we should allow people to override rules in `//go_repositories.bzl` in WORKSPACE files. For example, if they want to apply a patch to some `go_repository`, they can redefine the rule with the same `name` in WORKSPACE. Then `//go_repositories.bzl` contains no manual edits.